common/management/commands/project_external_updates.py

```diff
@@ -0,0 +1,92 @@
+from django.core.management.base import BaseCommand
+from common.helpers.db import bulk_delete
+from common.helpers.github import fetch_github_info, get_owner_repo_name_from_public_url, \
+    get_repo_endpoint_from_owner_repo_name, get_repo_names_from_owner_repo_name
+from common.helpers.date_helpers import datetime_field_to_datetime
+from django.conf import settings
+import pytz
+import traceback
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        project_github_links = get_project_github_links()
+        for github_link in project_github_links:
+            if github_link.link_project.is_searchable:
+                try:
+                    handle_project_github_updates(github_link)
+                except:
+                    # Keep processing if we run into errors with a particular update
+                    print('Error processing ' + github_link.link_url)
+                    print(traceback.format_exc())
+                    pass
+
+
+def get_project_github_links():
+    from civictechprojects.models import ProjectLink
+    return ProjectLink.objects.filter(link_name='link_coderepo', link_url__icontains='github.com/')
+
+
+def handle_project_github_updates(project_github_link):
+    project = project_github_link.link_project
+    print('Handling updates for project {id} github link: {url}'.format(id=project.id, url=project_github_link.link_url))
+    last_updated_time = datetime_field_to_datetime(get_project_latest_commit_date(project_github_link.link_project))
+    owner_repo_name = get_owner_repo_name_from_public_url(project_github_link.link_url)
+
+    repo_names = get_repo_names_from_owner_repo_name(owner_repo_name)
+    raw_commits = []
+    for repo_name in repo_names:
+        repo_url = get_repo_endpoint_from_owner_repo_name(repo_name, last_updated_time)
+        print('Ingesting: ' + repo_url)
+        repo_info = fetch_github_info(repo_url)
+        if repo_info is not None and len(repo_info) > 0:
+            repo_display_name = repo_name[0] + '/' + repo_name[1]
+            raw_commits = raw_commits + list(map(lambda commit: [repo_display_name, commit], repo_info))
+
+    if len(raw_commits) > 0:
+        # Take the most recent top X commits
+        raw_commits.sort(key=lambda commit: commit[1]['commit']['author']['date'], reverse=True)
+        raw_commits = raw_commits[:settings.MAX_COMMITS_PER_PROJECT]
```

Comment:
Out of curiosity if we didn't set MAX_COMMITS in the settings what'd happen here? Might be worth adding a hardcoded value to use in case max commits doesn't exist?

Reply:
We do, the default is 30